### PR TITLE
#552 | Redesign Transaction Page

### LINKED ISSUE #552: Transaction PAge

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #552

**Note:** In low resolution the tabs are being pushed out of the screen because they don't have a proper design (which is smaller than it is now). But this is a job for another issue since it touches more files than just the Transaction Page.

## Description

This PR adds the new redesign for the transaction page.

## Test scenarios
- Normal: https://deploy-preview-562--teloscan-redesign.netlify.app/tx/0x6802543121a72c26556c58641ea041ffbcea2df09378158bf88543b0fb5623a0
- Deposit: https://deploy-preview-562--teloscan-redesign.netlify.app/tx/0x8d8b484430aee3b955aae8b2529f672ab63f9cc4ccba377bb7789283470e2558
- Names (function and contract): https://deploy-preview-562--teloscan-redesign.netlify.app/tx/0xea418287717042aac0bb87af0eb9685dab89178db5c87cde92a94d0c9b13c059
- Contract Deployment: https://deploy-preview-562--teloscan-redesign.netlify.app/tx/0xe9be6d8aa21ae263e604c749cb05d3032ffc9ad9254140617e8780facedaa8ea

## Screenshots
image
image

image

## Checklist:

-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have cleaned up the code in the areas my change touches
-   [ ] My changes generate no new warnings
-   [ ] Any dependent changes have been merged and published in downstream modules
-   [ ] I have checked my code and corrected any misspellings
-   [ ] I have removed any unnecessary console messages
-   [ ] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [ ] I have tested for mobile functionality and responsiveness
-   [ ] I have added appropriate test coverage

### CODE REVIEW

File: src/components/GasLimitAndUsage.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+// import { useI18n } from 'vue-i18n';
```
Comment (donnyquixotic) [inline src/components/GasLimitAndUsage.vue:3]:
```suggestion
```

File: src/components/GasLimitAndUsage.vue [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import { BlockData } from 'src/types';
+import { EvmTransactionParsed } from 'src/antelope/types';
+import { useI18n } from 'vue-i18n';
+
+// const { t: $t } = useI18n();
```
Comment (donnyquixotic) [inline src/components/GasLimitAndUsage.vue:9]:
```suggestion
```

File: src/components/GasLimitAndUsage.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        </q-tooltip>
+    </span>
+</div>
+</template>
+
+<style lang="scss">
+.c-gas-limit {
+    //
+}
+</style>
```
Comment (donnyquixotic) [inline src/components/GasLimitAndUsage.vue:72]:
```suggestion
```
can remove empty class

File: src/components/MethodField.vue [hunk 4]
```diff
@@ -65,22 +70,22 @@ const setValues = async () => {
...
         !props.trx.parsedTransaction
         && props.trx.to === ZERO_ADDRESSES
-        && props.trx.value
+        && propValue.value
         && parseInt(props.trx.gasPrice as string) === 0
     ) {
         nativeTooltipText.value = $t('pages.transactions.native_withdraw_tooltip');
         methodName.value = $t('pages.transactions.withdraw_action_name');
-    } else if (!props.trx.parsedTransaction && props.trx.input === '0x' && props.trx.value) {
+    } else if (!props.trx.parsedTransaction && props.trx.input === '0x' && propValue.value) {
```
Comment (donnyquixotic) [inline src/components/MethodField.vue:86]: we reference `0x` here and `0x0` on line 63, are these handling distinct cases or can we use `ZERO_ADDRESSES` for these?

File: src/types/Evm
Comment (donnyquixotic) [inline src/types/Evm:1]: can remove empty file

**Review by donnyquixotic — CHANGES_REQUESTED** [review]

**Comment by donnyquixotic** [comment]
@Viterbo Can you fix the footer anchoring for the tx page?

**Comment by Viterbo** [comment]
> @Viterbo Can you fix the footer anchoring for the tx page?

That code was written by @ezra-sg and is correct. Look at the answer from the indexer:
image

**Review by donnyquixotic — APPROVED** [review]

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**